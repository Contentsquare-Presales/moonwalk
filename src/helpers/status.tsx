const STATUS_MESSAGES = [
  "Unknown Status",
  "Launch is GO",
  "Launch is NO-GO",
  "Launch was a success",
  "Launch failed",
];

export const getStatusMessage = (statusNumber: number) =>
  STATUS_MESSAGES[statusNumber] || "Unknown Status";
