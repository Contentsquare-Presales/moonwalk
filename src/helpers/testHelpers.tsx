import React, { ReactElement } from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components/native";

import theme from "../theme";

export const renderWithTheme = (component: ReactElement) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);