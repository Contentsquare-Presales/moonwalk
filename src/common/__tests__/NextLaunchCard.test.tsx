import React from "react";
import { NextLaunchCard } from "../";
import { renderWithTheme } from '../../helpers/testHelpers'

describe("NextLaunchCard", () => {
  it("renders correctly with an image", () => {
    const tree = renderWithTheme(<NextLaunchCard imgUrl="https://example.com/img.jpg" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly without an image", () => {
    const tree = renderWithTheme(<NextLaunchCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
