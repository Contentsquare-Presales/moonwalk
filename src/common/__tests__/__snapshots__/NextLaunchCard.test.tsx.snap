// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextLaunchCard renders correctly with an image 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#00adf1",
        "borderRadius": 40,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginBottom": 40,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 16,
        "overflow": "hidden",
        "paddingBottom": 40,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 40,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://example.com/img.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0.9,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
  />
</View>
`;

exports[`NextLaunchCard renders correctly without an image 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#00adf1",
        "borderRadius": 40,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginBottom": 40,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 16,
        "overflow": "hidden",
        "paddingBottom": 40,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 40,
      },
    ]
  }
/>
`;
